src/main.rs

```diff
@@ -434,36 +435,50 @@ fn dice_roll_result_text_ui(
     }
 }
 
+#[derive(Parser, Debug)]
+#[command(author, version, about, long_about = None)]
+struct Args {
+    #[arg(short, long, default_value_t = 0)]
+    world_seed: u64,
+
+    #[arg(short, long, default_value_t = 1)]
+    env_seed: u64,
+}
+
 fn main() {
     let number_of_players = 2;
 
-    let mut world_seed: u64 = 0;
-    let mut env_seed: u64 = 0;
-
-    // Get cli resource specs, else generate new random world and env seeds.
-    // TODO: implement CliResource handling
-
-    if world_seed == 0 || env_seed == 0 {
-	
-	let mut key = [0u8; 16];
-	OsRng.fill_bytes(&mut key);
-
-	// If one, or the other is set, only generate for the unset one.
-	// This will allow easier testing later, for fixed world random env_seed.
-	// Or for specific AI testing, fixed env_seed but random world. 
-	if world_seed == 0 {
-	    world_seed = OsRng.next_u64();
-	}
-	if env_seed == 0 {
-	    world_seed = OsRng.next_u64();
-	}
+    let mut args = Args::parse();
+
+    if args.world_seed == 0 || args.env_seed == 0 {
+        let mut key = [0u8; 16];
+        OsRng.fill_bytes(&mut key);
+
+        // If one, or the other is set, only generate for the unset one.
+        // This will allow easier testing later, for fixed world random env_seed.
+        // Or for specific AI testing, fixed env_seed but random world.
+        if args.world_seed == 0 {
+            args.world_seed = OsRng.next_u64();
+        }
+        if args.env_seed == 0 {
+            args.env_seed = OsRng.next_u64();
+        }
     }
 
+    // Source of randomness for the game
+    let prng_resource = tiered_prng::PrngResource {
+        world_seed: args.world_seed,
+        env_seed: args.env_seed,
+    };
+
+    // Generate game map
+    let map = generate_board(number_of_players, get_randomness(prng_resource.world_seed));
+
     App::new()
-    // PRNG setup
-	.insert_resource(tiered_prng::PrngResource{ world_seed: world_seed, env_seed: env_seed})
-    // Plugins
-	.add_plugin(tiered_prng::PrngPlugin) // Adds Prng based resources for subcomponents
+        // PRNG setup
+        .insert_resource(prng_resource)
```

Comment:
Hmm, this would likely cause problems, as the prng engine instance wouldn't be available.

Reply:
Yes, order here is going to need to be tightly controlled. I'd be concerned about a future situation where we retire a resource, but need to keep that offset in the stream so that later prng resource types keep their fixed offset. 

I do agree this would work fine, but I want to keep these prng resource handling within a plugin, so that any changes there get an extra look to ensure consistent prng handling, also the ordering of the resources reading from the top prng stream is going to need to stay highly strict.